src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory_Methods.vb

```diff
@@ -242,45 +242,94 @@ Namespace Microsoft.CodeAnalysis.Semantics
+            ' we'll end up with two ISingleVariableDeclarations that have the same syntax node. So, we group by VariableDeclaratorSyntax
+            ' to put x and y in the same IMutliVariableDeclaration
+            Dim groupedDeclarations = declarations.GroupBy(Function(declaration)
+                                                               If declaration.Kind = BoundKind.LocalDeclaration AndAlso
+                                                                  declaration.Syntax.IsKind(SyntaxKind.ModifiedIdentifier) Then
+                                                                   Debug.Assert(declaration.Syntax.Parent.IsKind(SyntaxKind.VariableDeclarator))
+                                                                   Return declaration.Syntax.Parent
+                                                               Else
+                                                                   Return declaration.Syntax
+                                                               End If
+                                                           End Function)
 
+            Dim builder = ArrayBuilder(Of IVariableDeclaration).GetInstance()
+            For Each declarationGroup In groupedDeclarations
+                Dim first = declarationGroup.First()
+                Dim singleDeclarations As ImmutableArray(Of ISingleVariableDeclaration) = Nothing
+                Dim initializer As IVariableInitializer = Nothing
+                If first.Kind = BoundKind.LocalDeclaration Then
+                    singleDeclarations = declarationGroup.Cast(Of BoundLocalDeclaration).SelectAsArray(AddressOf GetSingleVariableDeclaration)
+
+                    ' The initializer we use for this group is the initializer attached to the last declaration in this declarator, as that's
+                    ' where it will be parsed in an error case.
+                    ' Initializer is only created if it's not the array initializer for the variable. That initializer is the initializer
+                    ' of the SingleVariableDeclaration child.
+                    Dim last = DirectCast(declarationGroup.Last(), BoundLocalDeclaration)
+                    If last.InitializerOpt IsNot Nothing AndAlso last.InitializerOpt IsNot last.ArrayCreationOpt Then
+                        Debug.Assert(TypeOf last.Syntax Is ModifiedIdentifierSyntax)
+                        Dim initializerValue As IOperation = Create(last.InitializerOpt)
+                        Dim declaratorSyntax = DirectCast(last.Syntax.Parent, VariableDeclaratorSyntax)
+                        Dim initializerSyntax As SyntaxNode = declaratorSyntax.Initializer
+
+                        ' As New clauses with a single variable are bound as BoundLocalDeclarations, so adjust appropriately
                         Dim isImplicit As Boolean = False
-                        If initializerSyntax Is Nothing Then
+                        If initializerSyntax Is Nothing AndAlso
```

Comment:
>initializerSyntax Is Nothing AndAlso [](start = 27, length = 36)

I think we should simply use InitializedByAsNew property to figure out which syntax to grab.

Reply:
Done.

---
In reply to: [146973501](https://github.com/dotnet/roslyn/pull/22842#discussion_r146973501) [](ancestors = 146973501)